man/man1/pkcsstats.1.in

```diff
@@ -0,0 +1,141 @@
+.RB [ OPTIONS ]
+.
+.PP
+.B pkcsstats
+.BR \-\-help | \-h
+.br
+
+.SH DESCRIPTION
+Displays mechanism usage statistics for openCryptoki. Usage statistics are
+collected by openCryptoki on a per user basis. For each user, mechanism
+usage is counted per configured slot and mechanism. For each mechanism a set of
+counters exist, one for each cryptographics strength of the cryptographic
+key used with the mechanism.
+.PP
+The available strengths are defined in the strength configuration file
+\fB/etc/opencryptoki/strength.conf\fP. Supported strengths are 112, 128, 192,
+and 256 representing the corresponding strength in bits.
+The strength configuration file defines how the strength is determined for the
+various key types. A strength of zero is used to count those mechanisms that
+do not use a key, or where the key strength is less than 112 bits.
+.PP
+.B Note:
+The strength does not specify the cryptographic strength of the mechanism, but
+the cryptographic strength of the key used with the mechanism (if any).
+For example, usage of mechanism CKM\_SHA256 is reported under strength 0,
+because no key is used with this mechanism. However, usage of mechanism
+CKM\_AES\_CBC is reported under strength 128, 192, or 256, dependent on the
+cryptographic size of the AES key used with it (and the definitions in the
+strength configuration file).
+.PP
+Statistics collection is enabled by default. It can be disabled and configured
+in the openCryptoki configuration file
+\fB/etc/opencryptoki/opencryptoki.conf\fP.
+By default only explicit mechanism usage statistics from PKCS#11 applications
+are collected.
+.PP
+Optionally, implicit mechanism usage statistics can be collected, where
+additional mechanisms are specified in mechanism parameters. For example,
+RSA\-PSS or RSA\-OAEP allow to specify a hash mechanism and a mask generation
+function (MGF) in the mechanism parameter. ECDH allows to specify a key
+derivation function (KDF) in the mechanism parameter. The PBKDF2 mechanism
+allow to specify a pseudo random function (PRF) in the mechanism parameter.
+.PP
+Also optionally, opencryptoki\-internal mechanism usage statistics can be
+collected. This collects usage statistics for crypto operations used internally
+for pin handling and encryption of private token objects in the data store.
+.PP
+.B Note:
+Implicit or internal mechanism usage can not be distinguished from explicit
+mechanism usage of PKCS#11 applications in the displayed statistics.
+.PP
+Statistics are collected in a POSIX shared memory segment per user. This shared
+memory segment contains all counters for all configured slots, mechanisms, and
+strengths. The shared memory segments are named
+\fBvar.lib.opencryptoki_stats_<uid>\fP, where \fBuid\fP is the numeric user\-id
+of the user the statistics belong to. The shared memory segmenst are
+automatically created for a user on the first attempt to colletc statistics
+(when not already existent). The shared memory segmenst can be deleted using
```

Comment:
to collect ...   and again segmenst -> segments

Reply:
fixed